target_chains/starknet/contracts/src/pyth.cairo

```diff
@@ -1,29 +1,384 @@
+use core::array::ArrayTrait;
+use core::fmt::{Debug, Formatter};
+use super::byte_array::ByteArray;
+use super::util::UnwrapWithFelt252;
+
+pub use pyth::{Event, PriceFeedUpdateEvent};
+
 #[starknet::interface]
-trait IPyth<T> {
-    fn get_name(self: @T) -> felt252;
-    fn set_name(ref self: T, name: felt252);
+pub trait IPyth<T> {
+    fn latest_price_info(self: @T, price_id: u256) -> PriceInfo;
```

Comment:
metacomment about this api: in some target chains we have two separate functions to fetch price and ema. I'm not sure which one is better, there is also a question of efficiency.

we also call the primitive one `unsafe` to make sure people do recency check or use `get_{ema_,}price_no_older_than`. I think would be nice to have `unsafe` and  `no_older_than` here too. (The default one with a valid period seconds was actually not a good choice so please don't add it)

Reply:
Changed to `get_price_unsafe` and `get_ema_price_unsafe` and added `publish_time != 0` check, similar to the ethereum contract. We'll add more methods later.